x-pack/plugins/uptime/server/lib/requests/search/fetch_chunk.ts

```diff
@@ -6,32 +6,47 @@
 
 import { refinePotentialMatches } from './refine_potential_matches';
 import { findPotentialMatches } from './find_potential_matches';
-import { ChunkFetcher, ChunkResult } from './monitor_summary_iterator';
 import { QueryContext } from './query_context';
+import { getHistogramInterval } from '../../helper/get_histogram_interval';
+import { getHistogramForMonitors } from '../get_monitor_states';
+import { MonitorSummary } from '../../../../common/runtime_types/monitor';
 
 /**
  * Fetches a single 'chunk' of data with a single query, then uses a secondary query to filter out erroneous matches.
  * Note that all returned data may be erroneous. If `searchAfter` is returned the caller should invoke this function
  * repeatedly with the new searchAfter value as there may be more matching data in a future chunk. If `searchAfter`
  * is falsey there is no more data to fetch.
  * @param queryContext the data and resources needed to perform the query
- * @param searchAfter indicates where Elasticsearch should continue querying on subsequent requests, if at all
- * @param size the minimum size of the matches to chunk
+ *  * @param size the minimum size of the matches to chunk
+ * @param index where to start for query
+
  */
-export const fetchChunk: ChunkFetcher = async (
-  queryContext: QueryContext,
-  searchAfter: any,
-  size: number
-): Promise<ChunkResult> => {
-  const { monitorIds, searchAfter: foundSearchAfter } = await findPotentialMatches(
-    queryContext,
-    searchAfter,
-    size
+export const fetchChunk = async (queryContext: QueryContext, size: number, index: number) => {
+  const { monitorIds, totalMonitors } = await findPotentialMatches(queryContext, size, index);
+
+  const minInterval = getHistogramInterval(
+    queryContext.dateRangeStart,
+    queryContext.dateRangeEnd,
+    // 12 seems to be a good size for performance given
+    // long monitor lists of up to 100 on the overview page
+    12
```

Comment:
If we're hard coding the size to 12 this function shouldn't take `size` as a parameter any longer.

A change like this needs new benchmarks to back it up, including the pathological cases where over-searching matters. 

That said, I do think the new `timespan` field has probably reduced the likelihood of the false positives that the pre-buffering solves, so maybe we can go from 1k to 500 or so.